### ISSUE TITLE
Trigger :type-mismatch when accessing members of nil

### ISSUE THREAD

**Issue by knubie** [clj-kondo/clj-kondo#2249]
**Is your feature request related to a problem? Please describe.**
In clojure(script) I am pretty sure that `(. nil foo)` or `(. nil -foo)` will always raise an exception, so I think it would be good to include it in the `:type-mismatch` linter. It would also catch cases like `(.-foo (when foo bar))`.

**Describe the solution you'd like**
Member access on `nil` or `(tu/union-type :nil ...))` (e.g. with `clojure.core/when`) should trigger the `:type-mismatch` linter.

I would be open to submitting a PR for this if someone could point me in the right direction.

**Comment by borkdude** [comment]
Is this a theoretical example or something you stumbled upon in real life coding? Who would write `(.foo nil)`?

**Comment by knubie** [comment]
> Is this a theoretical example or something you stumbled upon in real life coding? Who would write (.foo nil)?

The `(.foo nil)` is theoretical, but I stumbled upon this in my real life code base with `(.foo (when foo bar))`, although the real life example was more obscured. Something like this:

```clj
(let [foo (when bar ...)]
  ;; Several lines of code...
  (.doSomething foo))
```

My code is probably just poorly written, but I still think it would be helpful to have this type checked.

Another nil check that might be worth adding is `(name nil)`.

**Comment by borkdude** [comment]
The above won't be caught by clj-kondo currently since it assumes that when something can be correct, no warning should be issued (to prevent false positives). As such I don't think there is much we can do here.
